api/service/legacysync/downloader/client.go

```diff
@@ -18,9 +19,11 @@ type Client struct {
 }
 
 // ClientSetup setups a Client given ip and port.
-func ClientSetup(ip, port string) *Client {
+func ClientSetup(ip, port string, withBlock bool) *Client {
 	client := Client{}
-	client.opts = append(client.opts, grpc.WithInsecure())
+	if withBlock {
+		client.opts = append(client.opts, grpc.WithInsecure(), grpc.WithBlock())
```

Comment:
grpc.WithInsecure() will not be added if withBlock is false.

Reply:
Fixed, (it happened somehow during rebase and fixing conflicts)